core/src/main/java/org/testcontainers/containers/ContainerWatcher.java

```diff
@@ -0,0 +1,33 @@
+package org.testcontainers.containers;
+
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class ContainerWatcher extends TestWatcher {
```

Comment:
this will break things and not how JUnit expects rules to work.
I would rather add "restart" to the VncRecorder